### ISSUE TITLE
Mentions Addon

### ISSUE THREAD

**Issue by Genius398** [ScratchAddons/ScratchAddons#1780]
This addon will alert you if someone will mention you with "@" (in Scratch). Eg: If a user mentions you on someone's profile/project/studio, the message will come as "@[username here] has mentioned you in a comment in [project/studio/profile name here]"

Full example:

@griffpatch has mentioned you in a comment in Robot Destructor. (Project Example)
@griffpatch has mentioned you in a comment in Military | The Army. (Studio Example)
@griffpatch has mentioned you in a comment in @World_Language's profile. (Project Example)

**Comment by Cool-Scratcher** [comment]
I like this idea. However, I think it could make spamming easier (mentioning hundreds of usernames). Therefore, I think we should have a feature where you can mute a certain person mentioning you, or make it so you only get alerts for certain users. This is because people will just start mentioning famous Scratchers to get their attention.

For example, if I was huntedskelly and I had Scratch Addons and PotatoAnimator also had it, I would not want to get alerts when a random person mentions me — I'd want to create a filter for a certain set of users _including him_ (and all my other friends).

If this was implemented, spam mentions would be solved.

**Comment by retronbv** [comment]
You do realize that when you did @`griffpatch` it pinged him? You have disturbed the beast O.O

**Comment by Genius398** [comment]
> I like this idea. However, I think it could make spamming easier (mentioning hundreds of usernames). Therefore, I think we should have a feature where you can mute a certain person mentioning you, or make it so you only get alerts for certain users. This is because people will just start mentioning famous Scratchers to get their attention.
> 
> For example, if I was huntedskelly and I had Scratch Addons and PotatoAnimator also had it, I would not want to get alerts when a random person mentions me — I'd want to create a filter for a certain set of users _including him_ (and all my other friends).
> 
> If this was implemented, spam mentions would be solved.

Yes, I forgot. We could have a list in which we could have all our followers (or specific people added by us). And only the messages/mentions by those people will show up in the Message popup

**Comment by Genius398** [comment]
> You do realize that when you did @`griffpatch` it pinged him? You have disturbed the beast O.O

Could not understand. Please explain (even if it was a joke)

**Comment by retronbv** [comment]
> > You do realize that when you did @`griffpatch` it pinged him? You have disturbed the beast O.O
> 
> Could not understand. Please explain (even if it was a joke)

You pinged griffpatch, any messages in this issue will now goto his GitHub notifications.

**Comment by RedGuy12** [comment]
Impossible. The addon would have to go through literally every comment on the website.

**Comment by Genius398** [comment]
> Impossible. The addon would have to go through literally every comment on the website.

Yes, I forgot. We could have a list in which we could have all our followers (or specific people added by us). And only the messages/mentions by those people will show up in the Message popup

**Comment by Genius398** [comment]
> Yup, this is impossible.

It's impossible until it's done


Yes, I forgot. We could have a list in which we could have all our followers (or specific people added by us). And only the messages/mentions by those people will show up in the Message popup

**Comment by pufferfish101007** [comment]
> > Yup, this is impossible.
> 
> It's impossible until it's done
> 
> Yes, I forgot. We could have a list in which we could have all our followers (or specific people added by us). And only the messages/mentions by those people will show up in the Message popup

You can't fetch comments made by a user, so it's still impossible.

**Comment by Genius398** [comment]
> > > Yup, this is impossible.
> > 
> > 
> > It's impossible until it's done
> > Yes, I forgot. We could have a list in which we could have all our followers (or specific people added by us). And only the messages/mentions by those people will show up in the Message popup
> 
> You can't fetch comments specifically made by a user, so it's still impossible.

I'll share the code soon

**Comment by griffpatch** [comment]
HOW DARE YOU PING ME... ROOARRRR! lol :P

(Seriously - I'm not a beast - Promise)

**Comment by RedGuy12** [comment]
No it is not in any way possible….

**Comment by Genius398** [comment]
DOn't close this thing

**Comment by TheColaber** [comment]
It just isn't possible and we have talked about it in the discord server as well. There is no API that could help for this kind of thing, and if we wanted to search every API, scratch would ban your IP from using their APIs.

**Comment by RedGuy12** [comment]
> DOn't close this thing

it's phisically impossible

**Comment by FunctionalMetatable** [comment]
> > DOn't close this thing
> 
> it's physically impossible

if only lefty's api will add comment searching..
